ui/src/components/tag/AddTagForm.js

```diff
@@ -0,0 +1,255 @@
+        } else {
+            this.state = {
+                newTagValue: '',
+                tagName: '',
+                tagValues: [],
+            };
+        }
+    }
+
+    inputChanged(key, evt) {
+        let value = '';
+        if (evt.target) {
+            value = evt.target.value;
+        } else {
+            value = evt ? evt : '';
+        }
+        this.setState({ [key]: value }, () =>
+            this.props.onUpdate(
+                this.state.tagName,
+                this.state.newTagValue,
+                this.state.tagValues
+            )
+        );
+    }
+
+    addTagValue() {
+        let name = this.state.newTagValue;
+        let tagValues = AppUtils.deepClone(this.state.tagValues);
+
+        if (!name) {
+            return;
+        }
+        let names = (name || '')
+            .replace(/[\r\n\s]+/g, ',')
+            .split(',')
+            .map((n) => n.trim())
+            .filter((n) => n);
+
+        for (let i = 0; i < names.length; i++) {
+            if (tagValues.indexOf(names[i]) === -1) {
+                tagValues.push(names[i]);
+            }
+        }
+
+        this.setState(
+            {
+                tagValues,
+                newTagValue: '',
+            },
+            () =>
+                this.props.onUpdate(
+                    this.state.tagName,
+                    this.state.newTagValue,
+                    this.state.tagValues
+                )
+        );
+    }
+
+    removeValue(idx) {
+        let tagValues = AppUtils.deepClone(this.state.tagValues);
+        tagValues.splice(idx, 1);
+        this.setState({ tagValues }, () =>
+            this.props.onUpdate(
+                this.state.tagName,
+                this.state.newTagValue,
+                this.state.tagValues
+            )
+        );
+    }
+
+    render() {
+        let tagValues = this.state.tagValues
+            ? this.state.tagValues.map((val, idx) => {
+                  let remove = this.removeValue.bind(this, idx);
+                  return (
+                      <StyledTagColor key={val} onClickRemove={remove}>
+                          <StyledAnchor style={StyledAnchorActiveInline}>
+                              {' '}
+                              {val}{' '}
+                          </StyledAnchor>
+                      </StyledTagColor>
+                  );
+              })
+            : '';
+
+        return (
+            <SectionsDiv autoComplete={'off'} data-testid='add-tag-form'>
+                <SectionDiv>
+                    <StyledInputLabel htmlFor='tag-name'>
+                        Tag Name
+                    </StyledInputLabel>
+                    <ContentDiv>
+                        <StyledInput
+                            id='tag-name'
+                            name='tag-name'
+                            placeholder='Enter New Tag Name'
+                            value={this.state.tagName}
+                            onChange={(evt) =>
```

Comment:
same here, this.inputChanged should be bind in constructor and pass it to this function to avoid creating anonymous functions on render